src/coreclr/jit/emit.h

```diff
@@ -277,6 +278,9 @@ struct insGroup
                                   // and the emitter should continue to track GC info as if there was no new block.
 #define IGF_HAS_ALIGN 0x0400      // this group contains an alignment instruction(s) at the end to align either the next
                                   // IG, or, if this IG contains with an unconditional branch, some subsequent IG.
+#define IGF_UPD_ICOUNT                                                                                                 \
```

Comment:
More generally, this will invalidate any and all `emitLocation`s captured before the update. They're also used for `USING_SCOPE_INFO` and unwinding, I wonder if the latter could explain the CI failures.

Reply:
It would be strange if that were the case only on osx which is the only pattern I can see, the other failures are weird things.

Reply:
I mean, the CI is red everywhere, e. g. in `coreclr Pri0 Runtime Tests Run windows x64 checked failed` we see:
```
    JIT\opt\Inline\tests\Inline_DetectChanges\Inline_DetectChanges.cmd [FAIL]
      Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
         at Program.Foo(Object o)
         at Program.Main()
```
Which does look a lot like an unwinding failure (-- EH code getting confused and not catching the exception). But this is just a guess.

Reply:
If I tried to change things I'd be guessing too. What I've done is what's is needed to achieve the desired result and satisfy the existing sanity checking assertions inside the runtime. If more needs to be done around liveness tracking then I'll need some direction from a person that knows how it works.

Reply:
I've done some experimenting and I can't see a cause or change the outcome. I've tried excluding prolog and epilog from removal and it has no effect. In discussions on discord we talked about whether the unwinder is being messed up somehow but I don't see how it can be when the unwind info is generated after all the assembly has been generated. Similarly any error handling has been generated into funclet instruction groups so there are no ranges to worry about. The jumps which are being removed are added after each basic block has been emitted into an IG so there shouldn't be any scope for liveness failures or registers being changed.

Reply:
I tried taking a spare bit in the `instrDesc` structure and setting it only on jmps emitted at the end of always jump marked basic blocks. Then used that to select if the jump can be removed so that nothing can be accidentally removed in finally blocks etc. It still breaks. I'm unable to continue, I have no nowhere to go. The exception handling seems to be broken by this change somehow but I don't know how or where without informed guidance. It's clear that there are benefits to this imo but I can't complete it from where it is currently.